### PR TITLE
ENH: integration of SegmentStatistics (issue #168, #161)

### LINKED ISSUE #168: Integrate with updated SegmentStatistics
It would be good to start integrating with the code being developed in #161 to see if there are any gaps and more fully test the new functionality.

### PR DESCRIPTION
**Pull request by che85**
fixes #168

* segmentStatisticsConfigButton added for configuring SegmentStatistics
  calculators
  * TODO: this should be moved somewhere else since in FourUp
          Quanitative the Measurements area disappears from main module
          widget

* for now just catching the exception and printing a message
* all quantities/units for which we don't have the code sequence
  definitions yet, won't be added to the output DICOM
  * TODO: add code sequences for handling quantities/units other than
          the currently supported ones only

We can merge this once the new SegmentStatistics additions has been merged to Slicer core

### CODE REVIEW

**Comment by fedorov** [comment]
@che85 quantities/units have been added in https://github.com/chribaue/Slicer/pull/2

**Comment by che85** [comment]
Alright. I will go ahead and test it

**Comment by chribaue** [comment]
#che85 When you're done testing, please let me know if you consider the plugin mechanism ready for a PR into Slicer.

**Comment by che85** [comment]
Loading a report back into Slicer after creation with QuantitativeReporting and integrated SegmentStatistics I am not getting the following:

image

**Comment by che85** [comment]
Need to investigate...

File: QuantitativeReporting.py [hunk 1]
```diff
@@ -991,65 +1031,33 @@ def generateJSON4DcmSR(self, dcmSegmentationFile, sourceVolumeNode):
...
-        return self.createCodeSequence("R-00317", "SRT", "Mean")
+    for key in [k for k in self.getNonEmptyKeys() if k not in ["Segment", "Labelmap.voxel_count"]]:
+      measurementInfo = self.getMeasurementInfo(key)
+      if measurementInfo:
+        item = dict()
+        item["value"] = str(self.statistics[segmentValue, key])
+        item["quantity"] = self._createCodeSequence(measurementInfo["DICOM.QuantityCode"])
+        item["units"] = self._createCodeSequence(measurementInfo["DICOM.UnitsCode"])
+        if measurementInfo.has_key("DICOM.UnitsCode"):
+          item["derivationModifier"] = self._createCodeSequence(measurementInfo["DICOM.UnitsCode"])
```
Comment (fedorov) [inline QuantitativeReporting.py]: this should be `DICOM.DerivationCode` instead of `DICOM.UnitsCode`

**Comment by che85** [comment]
@fedorov if you find time, please go ahead and test

**Comment by fedorov** [comment]
@che85 so to do that I just need to copy SegmentStatistics branch new files to the nightly folder, and then add the QR branch to the path? Anything else?

**Comment by che85** [comment]
That should work, yes.

**Comment by che85** [comment]
Merging this branch for next day testing. When loading SR from DICOM, duplicated table header names are enumerated. Units are displayed in headers and human readable units are displayed in tooltips.

image

**Pull request merged**